scripts/validate_proconnect_allowlist.py

```
@@ -0,0 +1,315 @@
+    """Whether a URL is an https link to service-public.gouv.fr (or a subdomain).
+
+    Subdomains are allowed because that is where the collectivité fiches live
+    (``lannuaire.service-public.gouv.fr/...``).
+    """
+    host = urlparse(url).hostname or ""
+    return urlparse(url).scheme == "https" and (
+        host == SERVICE_PUBLIC_HOST or host.endswith(f".{SERVICE_PUBLIC_HOST}")
+    )
+
+
+def _source_suffix(comment):
```

Ruff (0.16.1)

[warning] Missing return type annotation for private function `_source_suffix`

Add return type annotation: `str`

(ANN202)